src/v/kafka/server/protocol.cc

```diff
@@ -100,6 +100,13 @@ ss::future<> protocol::apply(rpc::server::resources rs) {
     return ss::do_until(
              [ctx] { return ctx->is_finished_parsing(); },
              [ctx] { return ctx->process_one_request(); })
+      .handle_exception([ctx](std::exception_ptr propogate) {
+          vlog(
+            klog.info,
+            "Auth failed on ip: {}",
```

Comment:
this is a generic location to print out exceptions that bubble up from the kafka layer. we should add connection information to it, but it is not specific to `Auth failed`.

Are we still going to pursue the logging of authorization (via the centralized solution using ctx.authorize) and authentication failures which I think we can intercept in the sasl handlers? That seemed like a fairly mechanical change that would improve logging across a bunch of places.

Reply:
> this is a generic location to print out exceptions that bubble up from the kafka layer. we should add connection information to it, but it is not specific to `Auth failed`.

Understood. I'll modify the message to be more clear.
 
> Are we still going to pursue the logging of authorization (via the centralized solution using ctx.authorize) and authentication failures which I think we can intercept in the sasl handlers? That seemed like a fairly mechanical change that would improve logging across a bunch of places.

I did not include the logging in ctx.authorize in this PR because I was concerned it would be a dup to what is printed in protocol.cc.

Reply:
in general it shouldn't be a duplicate. here an exception is bubbled up from within kafka and the connection is going to be forcefully shut down. when an authorization failure occurs we want to send back an error message to the client. it might be a duplicate in some cases, but i chose `produce` at random, and see that an error is returned without throwing an exception.

Reply:
if you haven't already considered it, you might try enabling sasl and then observing how authn/authz failures manifest in the log. my hunch is authn errors might show up in this central location with a thrown exception, but that authz may not.

Reply:
> but i chose `produce` at random, and see that an error is returned without throwing an exception.

In my tests so far all the authentication and authorization failures are caught before they reach protocol.cc such as incorrect scram-mech, incorrect creds, etc. I've had to forcefully throw an exception to get a failure to bubble far enough. What did you run?

Reply:
i didn't run anything only looking at the code. ultimately i don't think we should worry too much about it. i think adding context to every log message at the kafka layer could be useful. the spot in protocol.cc you chose is good, and then there is a big bang for the buck by hijacking ctx.authorize and the sasl authentication failure. then after that we can consider adding a contextual logger into the kakfa layer and replacing all calls to `klog` with that contextual logger.